testing-files/faultyThing.js

```diff
@@ -201,23 +112,97 @@ srv.start()
                     },
                 },
             },
-            (input) => {
+        },
+        events: {
+            failEvent: {
+                data: {
+                    type: "number",
+                },
+            },
+        },
+        id: "urn:uuid:560290c8-6490-4a58-99ca-eea9bc8c25d2",
+    })
+
+        .then((thing) => {
+            thing.writeProperty("display", "initialization string")
+            thing.writeProperty("wrongWritable", 15)
+
+            thing.setPropertyWriteHandler("wrongWritable", async () => {
+                console.log("Writing the old value.")
+                return 15
+            })
```

Comment:
@egekorkan Is this the correct use of a propertyHandler? I could observe two strange behaviors with this function: 

1. when trying to call `thing.writeProperty("wrongWritable"´, 15)` in the handler function the Thing logs "Writing the old value" about 40 times and then crashes when this handler is called. 

2. when nothing is returned e. g. `return` instead of `return 15` (without a call of `thing.writeProperty(...)`) the testbench receives `undefined` for second read and thus the test result is "wrong type" instead of "second read did not match write". This is strange because in this writeHandler nothing happend the original value should be unchanged.

Is this potentially another issue in node-wot?

Reply:
The correct way should be with resolve(). The value in resolve is the actual value written.

Reply:
Ah ok wenn writing `return` (without value) in an async function or just `resolve()` (without value) in a promise node-wot probably writes null thus the output was correct.